### PR TITLE
Add IThrowCollide

### PR DESCRIPTION
**Pull request by gradientvera**
- You can now cream pie people!
- Throwing an activated stunbaton at someone stuns them.

### CODE REVIEW

File: Resources/Prototypes/SoundCollections/desceration.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- type: soundCollection
+  id: desceration
```
Comment (RemieRichards) [inline Resources/Prototypes/SoundCollections/desceration.yml:2]: The word is desecration. 9 typos of this :P

Comment (Zumorica) [reply Resources/Prototypes/SoundCollections/desceration.yml:2]: blame SS13

File: Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+            return ReagentUnit.Zero;
+        }
+
+        public void ThrowCollide(ThrowCollideEventArgs eventArgs)
+        {
+            if (eventArgs.Target != Owner || !eventArgs.Thrown.TryGetComponent(out CreamPieComponent creamPie) || CreamPied) return;
+
+            CreamPied = true;
+            Owner.PopupMessage(Loc.GetString("You have been creamed by {0:theName}!", eventArgs.User));
```
Comment (RemieRichards) [inline Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:57]: this message seems... off lol.

Comment (Zumorica) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:57]: Also copying SS13 in this regard :cold_sweat: `H.visible_message("<span class='warning'>[H] is creamed by [src]!</span>", "<span class='userdanger'>You've been creamed by [src]!</span>")`
Any suggestions to make it better?

Comment (RemieRichards) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:58]: Your message doesn't match SS13's.

Yours is "You have been creamed by bob!", SS13's is "You have been creamed by the cream pie!"

Comment (Zumorica) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:58]: oh no, `eventArgs.User` is the person who creamed you, while `eventArgs.Thrown` is the cream pie itself in this case

File: Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Map;
+
+namespace Content.Shared.Interfaces.GameObjects.Components
+{
+    public interface IThrowCollide
```
Comment (RemieRichards) [inline Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs:8]: What is this interface?

It seems to be both being hit by things and htiting things?

Seems a bit messy (lots of "were we the thing hit or were we thrown" checks (`if(entity.Thrown==Owner)`))

File: Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs (hunk 2 above)
Comment (RemieRichards) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:58]: Yes, and you're using user. SS13 uses the pie.

File: Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs (hunk 3 above)
Comment (Zumorica) [reply Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs:8]: hmm, should I make this two different methods on the same interface?
One for when being hit, and one for when hitting something else

File: Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:58]: oh

Comment (Zumorica) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:58]: I blame BYOND's shitty syntax

File: Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs (hunk 3 above)
Comment (RemieRichards) [reply Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs:8]: I'd advocate for interfaces like this to be single use.

i.e. `IThrowCollisionHit` and `IThrowCollisionHitBy` (names up for discussion)

Comment (RemieRichards) [reply Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs:8]: Two methods is good, yes

File: Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs (hunk 2 above)
Comment (RemieRichards) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs]: Not really a syntax issue, you just misinterpreted byond's version of `this` :P

File: Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs (hunk 3 above)
Comment (Zumorica) [reply Content.Shared/Interfaces/GameObjects/Components/Interaction/IThrowCollide.cs:8]: Sounds good! Done.

File: Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    Wash();
+                    break;
+            }
+
+            return ReagentUnit.Zero;
+        }
+
+        public void HitBy(ThrowCollideEventArgs eventArgs)
+        {
+            if (eventArgs.Target != Owner || !eventArgs.Thrown.TryGetComponent(out CreamPieComponent creamPie) || CreamPied) return;
```
Comment (RemieRichards) [inline Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:55]: You can remove these target!=Owner, thrown!=Owner style checks now can't you?

Comment (Zumorica) [reply Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs:55]: yeah, I can. I forgot to change this--

**Pull request merged**